crates/bevy_pbr/src/ssao/mod.rs

```diff
@@ -0,0 +1,982 @@
+            .contains(TextureUsages::STORAGE_BINDING)
+        {
+            warn!("ScreenSpaceAmbientOcclusionPlugin not loaded. GPU lacks support: TextureFormat::R16Float does not support TextureUsages::STORAGE_BINDING.");
+            return;
+        }
+
+        if render_app
+            .world
+            .resource::<RenderDevice>()
+            .limits()
+            .max_storage_textures_per_shader_stage
+            < 5
+        {
+            warn!("ScreenSpaceAmbientOcclusionPlugin not loaded. GPU lacks support: Limits::max_storage_textures_per_shader_stage is less than 5.");
+            return;
+        }
+
+        render_app
+            .init_resource::<SsaoPipelines>()
+            .init_resource::<SpecializedComputePipelines<SsaoPipelines>>()
+            .add_systems(ExtractSchedule, extract_ssao_settings)
+            .add_systems(Render, prepare_ssao_textures.in_set(RenderSet::Prepare))
+            .add_systems(Render, prepare_ssao_pipelines.in_set(RenderSet::Prepare))
+            .add_systems(Render, queue_ssao_bind_groups.in_set(RenderSet::Queue))
+            .add_render_graph_node::<ViewNodeRunner<SsaoNode>>(
+                CORE_3D,
+                draw_3d_graph::node::SCREEN_SPACE_AMBIENT_OCCLUSION,
+            )
+            .add_render_graph_edges(
+                CORE_3D,
+                &[
+                    // PREPASS -> SCREEN_SPACE_AMBIENT_OCCLUSION -> MAIN_PASS
+                    bevy_core_pipeline::core_3d::graph::node::PREPASS,
+                    draw_3d_graph::node::SCREEN_SPACE_AMBIENT_OCCLUSION,
+                    bevy_core_pipeline::core_3d::graph::node::START_MAIN_PASS,
+                ],
+            );
+    }
+}
+
+/// Bundle to apply screen space ambient occlusion.
+#[derive(Bundle, Default)]
+pub struct ScreenSpaceAmbientOcclusionBundle {
+    pub settings: ScreenSpaceAmbientOcclusionSettings,
+    pub depth_prepass: DepthPrepass,
+    pub normal_prepass: NormalPrepass,
+}
+
+/// Component to apply screen space ambient occlusion to a 3d camera.
+///
+/// Screen space ambient occlusion (SSAO) approximates small-scale,
+/// local occlusion of _indirect_ diffuse light between objects, based on what's visible on-screen.
+/// SSAO does not apply to direct lighting, such as point or directional lights.
+///
+/// This darkens creases, e.g. on staircases, and gives nice contact shadows
+/// where objects meet, giving entities a more "grounded" feel.
+///
+/// # Usage Notes
+///
+/// Requires that you add [`ScreenSpaceAmbientOcclusionPlugin`] to your app,
+/// and add the [`DepthPrepass`] and [`NormalPrepass`] components to your camera.
+///
+/// It strongly recommended that you use SSAO in conjunction with
+/// TAA ([`bevy_core_pipeline::experimental::taa::TemporalAntiAliasSettings`]).
+/// Doing so greatly reduces SSAO noise.
+#[derive(Component, ExtractComponent, Reflect, PartialEq, Eq, Hash, Clone, Default)]
+#[reflect(Component)]
+pub enum ScreenSpaceAmbientOcclusionSettings {
```

Comment:
This should probably be a struct in preparation for future configuration. `SceenSpaceAmbientOcclusionSettings::quality_high()` seems like the right replacement. Or alternatively we could just internalize this whole enum as `SsaoQuality` or something.